Enable Node Tuning Operator in HyperShift

dagrayvid opened the pull request:
**What this PR does / why we need it**:
This PR enables the Node Tuning Operator to run in HyperShift environments. The Node Tuning Operator (NTO) manages the default TuneD profiles applied to worker nodes, and enables admins managing Hosted Clusters to apply custom TuneD profiles to the nodes in their NodePools.  The operator will run in the hosted control plane namespace of the management cluster. The operand (containerized TuneD daemon) will continue to run as a Daemonset on all of the worker nodes, so that it can apply the tunings to the underlying system. 

This PR is "phase 1" of enabling NTO. It will enable NTO to be used for applying basic tunings like sysctls. There will be a follow-up PR after this has merged to enable NTO to also set kernel boot parameters calculated by TuneD.

This PR contains the following changes, which are also outlined in this [design document](https://docs.google.com/document/d/1G9uu_EBCu-X8OgEA5L0P1GZMcrASjQJ6qIYAJSS6NbY/edit?usp=sharing):

- Control Plane Operator changes:
  - Add NTO Deployment, Metrics ServiceMonitor, and Service to the set of control plane components created and reconciled by the control-plane-operator in the hosted control plane namespace.
  - Re-enable the NTO ClusterOperator to be created in the hosted cluster.
  - Add PerformanceProfile CRD to the set of manifests to omit from hosted clusters (until "phase 3", 4.13 or later).
- NodePool controller changes:
  - Add spec.TunedConfig field to NodePool API, containing a list of object references. This field will be used to specify a list of ConfigMaps containing Tuned object manifests.
  - Add code to reconcile the spec.TunedConfig field. For each NodePool, propagate any Tuned manifests from the ConfigMaps referenced in the spec.TunedConfig field into ConfigMaps in the hosted control plane namespace where NTO can read them.

**Which issue(s) this PR fixes** 
[PSAP-742](https://issues.redhat.com/browse/PSAP-742)

**Checklist**
- [x] Subject and description added to both, commit and PR.
- [x] Relevant issues have been referenced.
- [x] This change includes docs. 
- [x] This change includes unit tests.

derekwaynecarr commented on api/v1alpha1/nodepool_types.go:
```diff
@@ -134,6 +134,17 @@ type NodePoolSpec struct {
+	// Each ConfigMap must have a single key named "tuned" whose value is the
+	// JSON or YAML of a serialized Tuned.
+	// +kubebuilder:validation:Optional
+	TunedConfig []corev1.LocalObjectReference `json:"tunedConfig,omitempty"`
```
this field and `Config` are examples of Linux only fields.  Do we want to start adding a Linux specific struct to encapsulate these fields?  It would potentially make Windows easier in future.

dagrayvid replied:
That seems reasonable to me... @csrwng what do you think? 

Is this something we want to address in this PR, or before this PR merges?

csrwng replied:
Let's address in a separate PR. @derekwaynecarr I can introduce the OS-specific struct in the v1beta1 API that I'm working on right now. We need to have a way to migrate the existing API given that it's already in use as is.

enxebre commented on hypershift-operator/controllers/nodepool/nodepool_controller.go:
```diff
@@ -811,6 +896,30 @@ func reconcileUserDataSecret(userDataSecret *corev1.Secret, nodePool *hyperv1.No
 }
 
+// jmencak
+func reconcileTunedConfigMap(tunedConfigMap *corev1.ConfigMap, nodePool *hyperv1.NodePool, config string) error {
```
This seems to be copied from reconcileTokenSecret which is a unique secret source of truth for Node release+config and watched by ignition server token secret controller. This new secret won't have any impact on the existing workflow.

You can see more details about the workflow here https://github.com/openshift/enhancements/blob/master/enhancements/hypershift/node-lifecycle.md

dagrayvid replied:
Resolving this as the hash / version logic has been removed for these Tuned configmaps.

enxebre commented on hypershift-operator/controllers/nodepool/nodepool_controller.go:
```diff
@@ -76,6 +77,12 @@ const (
+
+	nodePoolAnnotationCurrentTunedConfig        = "hypershift.openshift.io/nodePoolCurrentTunedConfig"
+	nodePoolAnnotationCurrentTunedConfigVersion = "hypershift.openshift.io/nodePoolCurrentTunedConfigVersion"
+	nodePoolAnnotationTargetTunedConfigVersion  = "hypershift.openshift.io/nodePoolTargetTunedConfigVersion"
```
nodePoolAnnotationTargetTunedConfigVersion seems unused

dagrayvid replied:
Good catch. This should now be fixed.

enxebre commented on control-plane-operator/controllers/hostedcontrolplane/nto/clusternodetuningoperator.go:
```diff
@@ -0,0 +1,211 @@
+			TargetPort: intstr.FromInt(60000),
+		},
+	}
+	//svc.Spec.Type = olmOperatorMetricsServiceDeepCopy.Spec.Type
```
please drop comment.

dagrayvid replied:
Thanks, I fixed this.

enxebre commented on go.mod:
```diff
@@ -29,6 +29,7 @@ require (
 	github.com/opencontainers/image-spec v1.0.2
 	github.com/openshift/api v0.0.0-20220525145417-ee5b62754c68
 	github.com/openshift/cluster-api-provider-agent/api v0.0.0-20220227135922-dd6353f609dc
+	github.com/openshift/cluster-node-tuning-operator v0.0.0-20220614214129-2c76314fb3cc
```
it'd be nicer to review splitting vendoring into it's own commit.

dagrayvid replied:
I split the vendoring into its own commit, thanks.

enxebre commented on hypershift-operator/controllers/nodepool/nodepool_controller.go:
```diff
@@ -567,6 +574,84 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 	}
 
+	// Validate tunedconfig input.
+	tunedConfig, err := r.getTunedConfig(ctx, nodePool, hcluster)
```
hcluster in header is unused.

enxebre replied:
can you please move all this logic up to where we do all the validation and conditions?

dagrayvid replied:
Thanks, I removed the unused hcluster and moved this logic up next to the other validation and status conditions.

enxebre commented on hypershift-operator/controllers/nodepool/nodepool_controller.go:
```diff
@@ -1246,6 +1395,13 @@ func defaultAndValidateConfigManifest(manifest []byte) ([]byte, error) {
 	case *v1alpha1.ImageContentSourcePolicy:
 	case *mcfgv1.KubeletConfig:
 	case *mcfgv1.ContainerRuntimeConfig:
+	case *tunedv1.Tuned:
```
this effectively enables tunedv1.Tuned as valid to be passed through .config knob which is not what we want. We need a different function to validate this.

dagrayvid replied:
Good point. I removed this and added a separate function specifically for the Tuned manifests.

enxebre commented on hypershift-operator/controllers/nodepool/nodepool_controller.go:
```diff
@@ -567,6 +574,84 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
+	log.Info("Target TunedConfig Version Hash Info:",
+		"target hash", targetTunedConfigVersionHash, "version", targetVersion, "tunedConfig", tunedConfig)
+
+	// Tuned ConfigMaps exist for each NodePool config/version and follow "prefixName-configVersionHash" naming convention.
```
I don't think this makes sense here, for .config is relevant because the whole payload including the OS is generated out of the .release + .config. In this case we just want to reconcile a config map that then the ntoperator watches.

enxebre commented:
Thanks @dagrayvid this is a great start. If the scope of this PR is to expose tune CRs in the NodePool API and reconcile those into the HCP namespace I don't think we need all the hash/version logic here.
I think we can start by:
- Validate input and reflect in a condition (as in the PR but without conflating with machine config validation).
- Add a simple reconcile via createOrUpdate function to reconcile the configMap.
- Have a condition to bubble up from the operator to reflect state: the tune was applied successfully / failure.

Let's make sure all this is fleshed out in the gdoc / kep so we can have further discussion there before proceeding and merging.

dagrayvid commented:
Thanks @enxebre for the prompt and thorough review. I think you're right that we don't need the hash/version logic for the Tuned ConfigMaps that are needed for the release / ignition config. I have simplified the PR quite a bit by removing all of this logic. Please let me know if these changes align with what you had in mind.

 I also updated the design gdoc quite a bit to be more up-to-date, though it's possible more detail is needed.

> Have a condition to bubble up from the operator to reflect state: the tune was applied successfully / failure.

I'm not sure what you have in mind here. NTO reports status in a few ways: report to the ClusterOperator about success/failure applying profiles with a count of how many Nodes if any, failed to apply the tuning, etc. The Profile CR objects status reflects the per-node success/failure applying the TuneD profiles. 

Should this status be reflected in some way in the NodePool status itself? I am pretty sure we don't want to block upgrades or progress on failed TuneD profile applications, but as both the ClusterOperator and Profile status is on the hosted cluster side, I suppose some status is needed. Any docs / examples you can point me to for other operators?

enxebre commented on hypershift-operator/controllers/nodepool/nodepool_controller.go:
```diff
@@ -811,6 +863,34 @@ func reconcileUserDataSecret(userDataSecret *corev1.Secret, nodePool *hyperv1.No
+
+	if tunedConfigMap.Data == nil {
+		tunedConfigMap.Data = map[string]string{}
+		tunedConfigMap.Annotations[TokenSecretTokenGenerationTime] = time.Now().Format(time.RFC3339Nano)
```
why do we need TokenSecretTokenGenerationTime?

dagrayvid replied:
My original thinking for keeping it there was that it would be nice to see when the ConfigMap was last updated. Outside of debugging purposes it's probably not necessary, so I have removed it.

enxebre commented on hypershift-operator/controllers/nodepool/nodepool_controller.go:
```diff
@@ -1186,6 +1266,73 @@ func (r *NodePoolReconciler) getConfig(ctx context.Context,
+		allConfigPlainText = append(allConfigPlainText, string(manifest))
+	}
+
+	// These configs are the input to a hash func whose output is used as part of the name of the user-data secret,
```
this comment does not makes sense here.

dagrayvid replied:
Thanks for catching this. I reworded this comment to be more in line with the current logic.

enxebre commented on hypershift-operator/controllers/nodepool/nodepool_controller.go:
```diff
@@ -1429,6 +1576,14 @@ func (r *NodePoolReconciler) enqueueNodePoolsForConfig(obj client.Object) []reco
 				break
 			}
 		}
+		for _, v := range nodePoolList.Items[key].Spec.TunedConfig {
```
having this here will needlessly run the loop even when the result was found in the loop above via .Spec.Config so we'll want to return earlier.

dagrayvid replied:
Good catch, I rearranged this part of the function to avoid unnecessary looping when we already know a NodePool needs to be reconciled.

enxebre commented:
>I'm not sure what you have in mind here. NTO reports status in a few ways: report to the ClusterOperator about success/failure applying profiles with a count of how many Nodes if any, failed to apply the tuning, etc. The Profile CR objects status reflects the per-node success/failure applying the TuneD profiles.

This should be good enough for starters so it's propagated from operator to HC via cvo.

dagrayvid commented on hypershift-operator/controllers/nodepool/nodepool_controller.go:
```diff
@@ -567,6 +574,84 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 		log.Info("Reconciled userData Secret", "result", result)
 	}
 
+	// Validate tunedconfig input.
```
Resolving this as the hash / version logic has been removed.

dagrayvid commented:
Thanks @enxebre. I have incorporated your latest suggestions and added a how-to doc on applying a simple tuned profile to the hosted nodes.

csrwng commented on hypershift-operator/controllers/nodepool/nodepool_controller.go:
```diff
@@ -811,6 +863,32 @@ func reconcileUserDataSecret(userDataSecret *corev1.Secret, nodePool *hyperv1.No
+	}
+
+	tunedConfigMap.Labels[tunedConfigMapLabel] = "true"
+	tunedConfigMap.Annotations[nodePoolAnnotation] = client.ObjectKeyFromObject(nodePool).String()
```
In addition to the annotation, you could also use a label for the nodepool, and then you can just use the label selector when finding configmaps that belong to the nodepool

dagrayvid replied:
I may be missing something, but I don't think we can use the full `namespace/name` of NodePools as a Label, as it contains a `/` which is invalid for a label value. 

I assume it's generally safe to assume that NodePool names will be unique across a hosted cluster, but I want to stick with the full namespace/name so that in "phase 2" when NTO is generating a MachineConfig for the NodePool controller to apply, NTO can label/annotate these MachineConfigs with the full namespace/name of the NodePool, so that the NodePool controller knows where to apply them.

csrwng replied:
This configmap ony lives in the control plane namespace, I don't think you need a full namespace/name, but if you wanted to you could just use a different separator. Anyway, it's just to make the querying of configmaps easier. If you don't think it's worth it, I'm ok with that too.

dagrayvid replied:
I went ahead and made the change you suggested, since NodePool name only (no namespace) is in-line with the https://github.com/openshift/hypershift/pull/1702/files.

csrwng commented on hypershift-operator/controllers/nodepool/nodepool_controller.go:
```diff
@@ -771,6 +812,17 @@ func (r *NodePoolReconciler) delete(ctx context.Context, nodePool *hyperv1.NodeP
 		}
 	}
 
+	// Delete any ConfigMap belonging to this NodePool i.e. TunedConfig ConfigMaps.
```
if you use labels, you could do `r.DeleteAllOf` instead of having a separate list function

csrwng commented on control-plane-operator/controllers/hostedcontrolplane/nto/clusternodetuningoperator.go:
```diff
@@ -0,0 +1,208 @@
+	return nil
+}
+
+func ReconcileClusterNodeTuningOperatorServiceMonitor(sm *prometheusoperatorv1.ServiceMonitor, ownerRef config.OwnerRef, clusterID string, metricsSet metrics.MetricsSet) error {
```
There's no need for an ownerRef. The createOrUpdate function should now take care of that. We acually need to remove it from other reconcile funcs.

dagrayvid replied:
Thanks, I removed these ownerRefs from the createOrUpdate functions here.

csrwng commented on hypershift-operator/controllers/nodepool/nodepool_controller.go:
```diff
@@ -1186,6 +1264,72 @@ func (r *NodePoolReconciler) getConfig(ctx context.Context,
+	scheme := runtime.NewScheme()
+	tunedv1.AddToScheme(scheme)
+
+	YamlSerializer := serializer.NewSerializerWithOptions(
```
use lower case for local var

dagrayvid replied:
Fixed.

csrwng commented on hypershift-operator/controllers/nodepool/nodepool_controller.go:
```diff
@@ -1186,6 +1264,72 @@ func (r *NodePoolReconciler) getConfig(ctx context.Context,
+		serializer.SerializerOptions{Yaml: true, Pretty: true, Strict: true},
+	)
+
+	cr, _, err := YamlSerializer.Decode(manifest, nil, nil)
```
How does defaulting happen with just decode/encode?

dagrayvid replied:
I renamed this func, since at least for now, we aren't defaulting the objects.

csrwng commented on hypershift-operator/controllers/nodepool/nodepool_controller.go:
```diff
@@ -811,6 +863,32 @@ func reconcileUserDataSecret(userDataSecret *corev1.Secret, nodePool *hyperv1.No
+	tunedConfigMap.Labels[tunedConfigMapLabel] = "true"
+	tunedConfigMap.Annotations[nodePoolAnnotation] = client.ObjectKeyFromObject(nodePool).String()
+
+	if tunedConfigMap.Data == nil {
```
it might be simpler to do:
```go

	if tunedConfigMap.Data == nil {
		tunedConfigMap.Data = map[string]string{}
        }
	tunedConfigMap.Data[tunedConfigKey] = tunedConfig
```

dagrayvid replied:
Certainly... Thanks, this should be fixed.

csrwng commented on control-plane-operator/controllers/hostedcontrolplane/nto/clusternodetuningoperator.go:
```diff
@@ -0,0 +1,208 @@
+	dep.Spec.Template.Spec.Volumes = []corev1.Volume{
+		{Name: "node-tuning-operator-tls", VolumeSource: corev1.VolumeSource{Secret: &corev1.SecretVolumeSource{SecretName: "node-tuning-operator-tls"}}},
+		{Name: "apiservice-cert", VolumeSource: corev1.VolumeSource{Secret: &corev1.SecretVolumeSource{
+			SecretName:  "performance-addon-operator-webhook-cert",
```
who creates this secret?

dagrayvid replied:
I removed this volume as it isn't needed until we are enabling PAO / PerformanceProfile controller, which will be at a later point when the CNF team has cycles. Thanks

csrwng reviewed (commented):
Thanks @dagrayvid 
Added some comments. Also I may have missed something but it's not clear to me how the Tuned resources get created in the guest cluster. I see you create a configmap per nodepool in the control plane namespace. But then not sure how they are picked up by the NTO.
Also, how does the NTO know to apply Tuned resources only to the machines of a given nodepool?

jmencak commented on docs/content/how-to/node-tuning.md:
````diff
@@ -0,0 +1,114 @@
+    ...
+    ```
+
+2. Now that the ConfigMap containg a Tuned manifest has been created and referenced in a NodePool, the Node Tuning Operator will sync the Tuned objects into the hosted cluster. You can check which Tuneds are defined, which profiles are set for each Node, and by labeling a Node with the label `tuned-1-node-label`, apply the custom profile we created to a Node.
````
typo: `s/containg/containing/`

dagrayvid replied:
Thanks, good catch.

jmencak commented on docs/content/how-to/node-tuning.md:
````diff
@@ -0,0 +1,114 @@
+    Example output:
+    ```
+    NAME                           TUNED            APPLIED   DEGRADED   AGE
+    worker-1                       tuned-1-profile   True      False      7m43s
````
Column alignment (extra space between `tuned-1-profile   True`

dagrayvid replied:
Thanks, good catch.

jmencak commented on docs/content/how-to/node-tuning.md:
````diff
@@ -0,0 +1,114 @@
+
+    As we can see, currently only the default Profiles are applied. If we label a Node in the hosted cluster, we should see that NTO applies the custom profile we defined earlier to that node.
+    ```
+    oc --kubeconfig="$HC_KUBECONFIG" label node/worker-1 tuned-1-node-label=""
````
I'm not sure we should encourage users to use custom labels on the nodes like this.  When the NodePool has the default `.spec.management.upgradeType = Replace` upgrade type, they will be wiped on NodePool upgrade.  Another option is to add a warning about this and perhaps recommend the `InPlace` upgrade type?  Feedback/comments from the HyperShift team very welcome here.

nit: shell expands `""` as an empty string, so this can be completely omitted.

dagrayvid replied:
Agreed, I removed the node label matching from the example Tuned CR, and added a node explaining this consideration underneath the CR. Now the example CR uses NodePool based matching,

jmencak commented on docs/content/how-to/node-tuning.md:
```diff
@@ -0,0 +1,114 @@
+# Manage node-level tuning with the Node Tuning Operator
+
+If you would like to set some node-level tuning on the nodes in your hosted cluster, you can use the [Node Tuning Operator](https://docs.openshift.com/container-platform/4.11/scalability_and_performance/using-node-tuning-operator.html). In HyperShift, node tuning can be configured by creating ConfigMaps which contain Tuned objects, and referencing these ConfigMaps in your NodePools. Currently Nod ...
```
`s/Tuned daemon/TuneD daemon/`

dagrayvid replied:
Thanks, fixed.

jmencak reviewed (commented):
Went through the docs, they look good David, thank you.  My only major concern is instructing users to use custom labels.  Either we should not do that (upgrades) or expand on the subject.

dagrayvid commented on hypershift-operator/controllers/nodepool/nodepool_controller.go:
```diff
@@ -771,6 +812,17 @@ func (r *NodePoolReconciler) delete(ctx context.Context, nodePool *hyperv1.NodeP
+	}
+	for k := range configMaps {
+		if err := r.Delete(ctx, &configMaps[k]); err != nil && !apierrors.IsNotFound(err) {
+			return fmt.Errorf("failed to delete secret: %w", err)
```
Thanks, good catch. I fixed this.

dagrayvid commented on docs/content/how-to/node-tuning.md:
```diff
@@ -0,0 +1,114 @@
+              summary=Custom OpenShift profile
+              include=openshift-node
+
+              [sysctl]   # Section with minimum changes
```
Removed this comment from the profile, thanks.

dagrayvid commented:
Thanks for the reviews. 

I think I addressed all the comments. I also added a commit which adds RBAC privileges so that NTO can read MachineSets, to map Nodes to NodePools. This commit can be removed once https://github.com/openshift/hypershift/pull/1702 is merged labelling Nodes with NodePool names, but I wanted to keep this branch in line with what works for https://github.com/openshift/cluster-node-tuning-operator/pull/390, which is currently using MachineSet annotations to map Nodes to NodePools.

jmencak commented on docs/content/how-to/node-tuning.md:
````diff
@@ -0,0 +1,97 @@
+    ...
+    ```
+
+2. Now that the ConfigMap containing a Tuned manifest has been created and referenced in a NodePool, the Node Tuning Operator will sync the Tuned objects into the hosted cluster. You can check which Tuneds are defined, which profiles are set for each Node, and by labeling a Node with the label `tuned-1-node-label`, apply the custom profile we created to a Node.
````
Please remove the relic `tuned-1-node-label`.  We no longer use it in this doc.

dagrayvid replied:
Done.

jmencak reviewed (commented):
Thank you for the changes, David!  Went through the docs again, only a few minor adjustments needed now AFAICS.

dagrayvid commented on docs/content/how-to/node-tuning.md:
```diff
@@ -0,0 +1,97 @@
+    kind: NodePool
+    metadata:
+    ...
+    name: nodepool-1
```
Fixed.

dagrayvid commented on docs/content/how-to/node-tuning.md:
```diff
@@ -0,0 +1,97 @@
+    ...
+    spec:
+    ...
+    tunedConfig:
```
Fixed.

dagrayvid commented on docs/content/how-to/node-tuning.md:
```diff
@@ -0,0 +1,97 @@
+
+If you would like to set some node-level tuning on the nodes in your hosted cluster, you can use the [Node Tuning Operator](https://docs.openshift.com/container-platform/4.11/scalability_and_performance/using-node-tuning-operator.html). In HyperShift, node tuning can be configured by creating ConfigMaps which contain Tuned objects, and referencing these ConfigMaps in your NodePools. Currently Nod ...
+
+1. Create a ConfigMap which contains a valid Tuned manifest and reference it in a NodePool. The example Tuned manifest below defines a profile which sets `vm.dirty_ratio` to 55, on Nodes which contain the Node label  `tuned-1-node-label` with any value. 
```
Done.

dagrayvid commented on control-plane-operator/controllers/hostedcontrolplane/nto/clusternodetuningoperator.go:
```diff
@@ -0,0 +1,208 @@
+						},
+					},
+				},
+			},
```
Done, thank you.

sjenning commented on hypershift-operator/controllers/nodepool/nodepool_controller.go:
```diff
@@ -529,6 +553,21 @@ func (r *NodePoolReconciler) reconcile(ctx context.Context, hcluster *hyperv1.Ho
 
+	tunedConfigMap := TunedConfigMap(controlPlaneNamespace, nodePool.Name)
+	if tunedConfig == "" {
+		if err := r.Delete(ctx, tunedConfigMap); err != nil && !apierrors.IsNotFound(err) {
```
I think this is causing our 404 delete detection to trigger.  To avoid unneeded API server calls, we do a Get first and if that returns IsNotFound, we skip the Delete.

dagrayvid replied:
Thank you Seth, good catch. I corrected this.

dagrayvid commented:
/retest

enxebre commented on control-plane-operator/controllers/hostedcontrolplane/nto/clusternodetuningoperator.go:
```diff
@@ -0,0 +1,209 @@
+func NewParams(hcp *hyperv1.HostedControlPlane, version string, images map[string]string, setDefaultSecurityContext bool) Params {
+	p := Params{
+		Images: Images{
+			NodeTuningOperator: images["cluster-node-tuning-operator"],
```
can you move cluster-node-tuning-operator into a const?

dagrayvid replied:
Thanks, I fixed this.

enxebre commented on control-plane-operator/controllers/hostedcontrolplane/nto/clusternodetuningoperator.go:
```diff
@@ -0,0 +1,209 @@
+}
+
+func ReconcileDeployment(dep *appsv1.Deployment, params Params) error {
+	dep.Spec.Replicas = utilpointer.Int32(1)
```
nit: don't need replicas, is set by setDefaults()

dagrayvid replied:
Thanks, I removed this line.

enxebre commented on control-plane-operator/controllers/hostedcontrolplane/nto/clusternodetuningoperator.go:
```diff
@@ -0,0 +1,209 @@
+	dep.Spec.Replicas = utilpointer.Int32(1)
+	dep.Spec.Selector = &metav1.LabelSelector{
+		MatchLabels: map[string]string{
+			"name": metricsServiceName,
```
why is this matchin metricsServiceName? should the variable be renamed?

dagrayvid replied:
Thanks, I cleaned up the naming to be more logical and consistent with standalone OCP (and made sure metrics were still working).

enxebre commented on control-plane-operator/controllers/hostedcontrolplane/nto/clusternodetuningoperator.go:
```diff
@@ -0,0 +1,209 @@
+	}
+	dep.Spec.Template.Labels = map[string]string{
+		"name": metricsServiceName,
+		"hypershift.openshift.io/control-plane-component": "node-tuning-operator",
```
please use hyperv1.ControlPlaneComponent

dagrayvid replied:
Done.

enxebre commented:
Thanks @dagrayvid, I Dropped some more minor comments, feel free to address them here or in a follow up PR.
/approve

dagrayvid commented:
Thanks @enxebre, I addressed your last few comments now.

dagrayvid commented:
/unhold

enxebre commented:
/lgtm
